lib/ace/ext/papeeria/tex_completer.coffee

```diff
@@ -86,37 +93,68 @@ define( (require, exports, module) ->
     meta: 'equation'
   )
   
-  init = (editor, bindKey) ->
+  #Specific for token's system of type in ace
+  #We saw such a realization in html_completions.js
+  istype = (token, type) -> 
+    return token.type.lastIndexOf(type) > -1
+  
+
+  init = (editor, bindKey) -> 
     HashHandler = require("ace/keyboard/hash_handler").HashHandler; 
     keyboardHandler = new HashHandler();
     keyboardHandler.addCommand(
       name: 'add item in list mode',
       bindKey: bindKey,
-      exec: (editor) ->
+      exec: (editor) ->  
         pos = editor.getCursorPosition()
         curLine = editor.session.getLine(pos.row);
+        intendCount = ContextHelper.getNestedListDepth(editor.session, pos.row)
         # it's temporary fix bug with added \item before \begin{itemize|enumerate}
-        if (editor.getSession().getContext(pos.row) == "list" && curLine.indexOf("begin") < pos.column) 
-          editor.insert("\n\t\\item ")
+        if ContextHelper.getContext(editor.session, pos.row) == LIST_STATE && curLine.indexOf("begin") < pos.column
+          editor.insert("\n" + "    ".repeat(intendCount) + "\\item ")
```

Comment:
Can you look if it is possible to get the tabsize setting from editor? https://github.com/ajaxorg/ace/wiki/Embedding-API